opcodes/src/lib.rs

```diff
@@ -58,6 +60,92 @@ pub enum Opcode {
     MULSSECP256K1,
 }
 
+pub fn map_opcode(opcode: u32) -> u32 {
+    match opcode {
+        1 => 1,     // LOAD32
+        2 => 17,    // STORE32
+        3 => 33,    // JAL
+        4 => 49,    // JALV
+        5 => 65,    // BEQ
+        6 => 81,    // BNE
+        7 => 97,    // IMM32
+        8 => 113,   // STOP
+        9 => 129,   // READ_ADVICE
+        10 => 145,  // LOADFP
+        11 => 161,  // LOADU8
+        12 => 177,  // LOADS8
+        13 => 193,  // STOREU8
+        16 => 16,   // FAIL
+        20 => 209,  // MEMCPY
+        100 => 100, // ADD32
+        101 => 101, // SUB32
+        102 => 102, // MUL32
+        109 => 109, // XOR32
+        111 => 111, // NE32
+        112 => 112, // MULHU32
+        113 => 114, // SRA32
+        114 => 115, // MULH32
+        115 => 116, // LTE32
+        116 => 117, // EQ32
+        117 => 118, // SLT32
+        118 => 119, // SLE32
+        200 => 200, // ADD
+        201 => 201, // SUB
+        202 => 202, // MUL
+        300 => 225, // WRITE
+        120 => 241, // KECCAKF
+        134 => 134, // COMBSECP256K1
+        _ => opcode,
+    }
+}
+
+pub fn map_opcode_to_field_value<F: PrimeField32>(opcode: u32) -> F {
+    F::from_canonical_u32(map_opcode(opcode))
+}
+
```

Comment:
It would be nice to add a little [property test](https://docs.rs/proptest/latest/proptest/) which checks that this is the inverse of `map_opcode`.